libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -155,6 +170,176 @@ static int32_t recvExact( MQTTTransportRecvFunc_t recvFunc,
+        packetSize += pPublishInfo->payloadLength;
+
+        /* Now that the "Remaining length" is known, recalculate the payload limit
+         * based on the size of its encoding. */
+        payloadLimit -= remainingLengthEncodedSize( packetSize );
+
+        /* Check that the given payload fits within the size allowed by MQTT spec. */
+        if( pPublishInfo->payloadLength > payloadLimit )
+        {
+            status = false;
+        }
+        else
+        {
+            /* Set the "Remaining length" output parameter and calculate the full
+             * size of the PUBLISH packet. */
+            *pRemainingLength = packetSize;
+
+            packetSize += 1U + remainingLengthEncodedSize( packetSize );
+            *pPacketSize = packetSize;
+        }
+    }
+
+    return status;
+}
+
+static MQTTStatus_t serializePublishCommon( const MQTTPublishInfo_t * pPublishInfo,
+                                            size_t remainingLength,
+                                            uint16_t packetIdentifier,
+                                            const MQTTFixedBuffer_t * const pFixedBuffer,
+                                            bool serializePayload,
+                                            size_t * const pHeaderSize )
+{
+    MQTTStatus_t status = MQTTSuccess;
+    uint8_t publishFlags = 0;
+    uint8_t * pIndex = NULL;
+    size_t minBufferSize = 0;
+
+    if( ( pFixedBuffer == NULL ) || ( pPublishInfo == NULL ) || ( packetIdentifier == 0U ) )
+    {
+        status = MQTTBadParameter;
+    }
+    else if( ( pPublishInfo->pTopicName == NULL ) || ( pPublishInfo->topicNameLength == 0U ) )
+    {
+        status = MQTTBadParameter;
+    }
+    else
+    {
+        /* Empty else MISRA 15.7 */
+    }
+
+    /* Calculate the minimum buffer size needed and check if the buffer is big enough. */
+    if( status == MQTTSuccess )
+    {
+        minBufferSize += remainingLength + 1U + remainingLengthEncodedSize( remainingLength );
+
+        /* Reduce the payload size if not needed to be serialized. */
+        if( serializePayload )
+        {
+            minBufferSize -= pPublishInfo->payloadLength;
+        }
+
+        /* Check if the buffer size is greater than the minimum buffer size. */
+        if( minBufferSize > pFixedBuffer->size )
+        {
+            status = MQTTNoMemory;
+        }
+    }
+
+    if( status == MQTTSuccess )
+    {
+        pIndex = pFixedBuffer->pBuffer;
+
+        /* The first byte of a PUBLISH packet contains the packet type and flags. */
+        publishFlags = MQTT_PACKET_TYPE_PUBLISH;
+
+        if( pPublishInfo->qos == MQTTQoS1 )
+        {
+            UINT8_SET_BIT( publishFlags, MQTT_PUBLISH_FLAG_QOS1 );
+        }
+        else if( pPublishInfo->qos == MQTTQoS2 )
+        {
+            UINT8_SET_BIT( publishFlags, MQTT_PUBLISH_FLAG_QOS2 );
+        }
+        else
+        {
+            /* Empty else MISRA 15.7 */
+        }
+
+        if( pPublishInfo->retain == true )
```

Comment:
Since we are including stdbool.h(C99) true and false should be defined.

Reply:
I brought this up here because it was also brought up in a [different PR](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/874#discussion_r414949314), which I tried to address with this [commit](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/874/commits/0d0d49cc31dcb7cbc2ca6613aa1d95263245a5cd). It hasn't been merged yet, though.

Reply:
ok. Thanks for bring this point. Fixed